examples/LLM_plus_GNN.py

```diff
@@ -0,0 +1,441 @@
+            inputs_embeds = self.word_embedding(
+                torch.tensor(input_ids).to(self.model.device))
+            inputs_embeds = torch.cat(
+                [bos_embeds, graph_embeds[i].unsqueeze(0), inputs_embeds],
+                dim=0)
+            batch_inputs_embeds.append(inputs_embeds)
+            batch_attention_mask.append([1] * inputs_embeds.shape[0])
+
+        # pad inputs_embeds
+        max_length = max([x.shape[0] for x in batch_inputs_embeds])
+        for i in range(batch_size):
+            pad_length = max_length - batch_inputs_embeds[i].shape[0]
+            batch_inputs_embeds[i] = torch.cat(
+                [pad_embeds.repeat(pad_length, 1), batch_inputs_embeds[i]])
+            batch_attention_mask[i] = [0
+                                       ] * pad_length + batch_attention_mask[i]
+
+        inputs_embeds = torch.stack(batch_inputs_embeds,
+                                    dim=0).to(self.model.device)
+        attention_mask = torch.tensor(batch_attention_mask).to(
+            self.model.device)
+
+        with torch.cuda.amp.autocast(dtype=torch.bfloat16):
+            outputs = self.model.generate(
+                inputs_embeds=inputs_embeds,
+                max_new_tokens=self.max_new_tokens,
+                attention_mask=attention_mask,
+                # do_sample=True,
+                use_cache=True  # IMPORTANT!
+            )
+        pred = self.tokenizer.batch_decode(outputs, skip_special_tokens=True)
+
+        return {
+            'pred': pred,
+            'label': samples['label'],
+            'question': samples['question'],
+            'desc': samples['desc'],
+        }
+
+    def print_trainable_params(self):
+        trainable_params = 0
+        all_param = 0
+        for _, param in self.named_parameters():
+            num_params = param.numel()
+
+            all_param += num_params
+            if param.requires_grad:
+                trainable_params += num_params
+
+        return trainable_params, all_param
+
+
+def main(since: float):
+    seed_everything(42)
+
+    dataset = WebQSPDataset()
+    idx_split = dataset.split_idxs
+
+    # Step 1: Build Node Classification Dataset
+    train_dataset = [dataset[i] for i in idx_split['train']]
+    val_dataset = [dataset[i] for i in idx_split['val']]
+    test_dataset = [dataset[i] for i in idx_split['test']]
+
+    train_loader = DataLoader(train_dataset, batch_size=4, drop_last=True,
+                              pin_memory=True, shuffle=True)
+    val_loader = DataLoader(val_dataset, batch_size=4, drop_last=False,
+                            pin_memory=True, shuffle=False)
+    test_loader = DataLoader(test_dataset, batch_size=4, drop_last=False,
+                             pin_memory=True, shuffle=False)
+
+    # Step 2: Build Model
+    model = GAT_LLAMA("meta-llama/Llama-2-7b-chat-hf")
+
+    # Step 3 Set Optimizer
+    params = [p for _, p in model.named_parameters() if p.requires_grad]
+    lr = 1e-5
```

Comment:
How about we make some of the hyperparams as script args: lr, epochs, grad_steps etc